src/adapter-manager.js

```diff
@@ -324,6 +333,102 @@ class AdapterManager extends EventEmitter {
   }
 
   /**
+   * @method loadAdapter
+   *
+   * Loads a single adapter.
+   */
+
+  async loadAdapter(adapterId) {
+    const adapterPath = path.join(__dirname,
+                                  config.get('adapterManager.path'),
+                                  adapterId);
+
+    // Skip if there's no manifest.
+    const manifestJson = path.join(adapterPath, 'manifest.json');
+    if (!fs.lstatSync(manifestJson).isFile()) {
+      console.error('manifest.json not found for adapter:', adapterId);
+      return;
+    }
+
+    // Read the manifest.
+    let data;
+    try {
+      data = fs.readFileSync(manifestJson);
+    } catch (err) {
+      console.error('Failed to read manifest.json for adapter:', adapterId);
+      console.error(err);
+      return;
+    }
+
+    let manifest;
+    try {
+      manifest = JSON.parse(data);
+    } catch (err) {
+      console.error('Failed to parse manifest.json for adapter:', adapterId);
+      console.error(err);
+      return;
+    }
+
+    // Get the enabled status, then delete the key from our copy of the
+    // adapter settings. All adapters left in the copy will be deleted
+    // from the settings after loading is finished.
+    const key = `adapters.${manifest.id}`;
+    let savedSettings = await Settings.get(key);
+    if (savedSettings) {
+      // Overwrite config and enablement values.
+      manifest.enabled = savedSettings.enabled;
+      manifest.config = Object.assign(manifest.config, savedSettings.config);
+    } else {
+      manifest.enabled = false;
+    }
+
+    // Update the settings database.
+    await Settings.set(key, manifest);
+
+    // Verify API version.
+    const apiVersion = config.get('adapterManager.api');
+    if (manifest.api !== apiVersion) {
+      console.error('API mismatch for adapter:', manifest.id,
+                    '\nExpected:', apiVersion, ' Actual:', manifest.api);
+      return;
+    }
+
+    // If this adapter is not explicitly enabled, move on.
+    if (!manifest.enabled) {
```

Comment:
This looks wrong to me, because 'enabled' doesn't come from the manifest but rather from the settings.

I think it would be clearer to just have a local enabled variable which is initialized from the manifest (if you intended that) and overridden by the setting.